h/api/nipsa/search.py

```diff
@@ -0,0 +1,83 @@
+import copy
+
+
+def nipsa_filter(query, userid=None):
```

Comment:
The name of this function is `nipsa_filter` and it doesn't really change the `query`, it just nests it inside the filtered query. What if instead this function just returned the filter and its caller could combine the query and the filter to make a filtered query?